updatesh_d/scriptutil.d

```diff
@@ -0,0 +1,184 @@
+module scriptutil;
+import core.sys.posix.unistd:geteuid;
+static import std.ascii;
+import std.typecons:Tuple,tuple;
+import std.datetime:Clock,SysTime;
+import std.digest.digest:digest;
+import std.digest.crc:crcHexString,CRC32;
+import std.file:mkdir,exists,isDir,File,dirSeparator;
+import std.process:executeShell;
+import std.string:strip,indexOf,munch;
+import std.array:join;
+import std.math:pow;
+import std.algorithm:max;
+import std.random:uniform;
+import std.conv:to;
+import std.stdio:writefln,stdout;
+// shamelessly nicked from http://www.mktemp.org/repos/mktemp/file/e4089ef40fd0/mkdtemp.c
+
+
+private string makeHash(immutable string path)
+{
+	return digest!CRC32(path).crcHexString ~ digest!CRC32((Clock.currTime.toISOExtString)).crcHexString;
+}
+
+private enum permittedCharacters= std.ascii.letters ~ std.ascii.digits;
+
+private string makeRandomAlphaNumeric(size_t numChars)
+{
+	string ret="";
+	foreach(i;0..numChars)
+	{
+		auto r=uniform(0,permittedCharacters.length);
+		ret~=to!char(r);
+	}
+	return ret;
+}
+
+
+enum TempDirMode
+{
+	createFile,
+	createDirectory,
+}
+
+enum MaxTries=100_000;
```

Comment:
Group constants at the top (as opposed to mixing them with function definitions). Their order doesn't matter anyway and it's good to have all configuration constants in a place.